### PR TITLE
Clarify usage of "arguments" and "parameters"

### PR DESCRIPTION
**Pull request by HertzDevil**
See crystal-lang/crystal#10374.

### CODE REVIEW

File: docs/syntax_and_semantics/blocks_and_procs.md [hunk 1]
```diff
@@ -132,19 +132,19 @@ end
...
+It's an error specifying more block parameters than the arguments yielded:
 
 ```crystal
 def twice
   yield
   yield
 end
 
-twice do |i| # Error: too many block arguments
+twice do |i| # Error: too many block parameters
```
Comment (straight-shoota) [inline docs/syntax_and_semantics/blocks_and_procs.md:143]: This change is not correct until the compiler error message is updated.

File: docs/syntax_and_semantics/default_and_named_arguments.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
-## Default values
+## Default arguments
```
Comment (straight-shoota) [inline docs/syntax_and_semantics/default_and_named_arguments.md:1]: I would leave this section title as is per https://github.com/crystal-lang/crystal/issues/10374#issuecomment-775140541 ff.

File: docs/syntax_and_semantics/blocks_and_procs.md (hunk 1 above)
Comment (HertzDevil) [reply docs/syntax_and_semantics/blocks_and_procs.md:143]: A _lot_ other error messages are also outdated, so I simply assume that they do not correspond to exact error messages and that we should revisit them later.

Comment (straight-shoota) [reply docs/syntax_and_semantics/blocks_and_procs.md:143]: Well the idea is to represent the actual error message.

Comment (straight-shoota) [reply docs/syntax_and_semantics/blocks_and_procs.md:143]: But I suppose we can apply this change in expectation that the compiler error message is going to change as well as suggested in crystal-lang/crystal#10374.

Comment (HertzDevil) [reply docs/syntax_and_semantics/blocks_and_procs.md:143]: The error messages are now covered by crystal-lang/crystal#10378 (although this particular one might also be affected by the changes in crystal-lang/crystal#10251 so I updated that as well).

**Review by straight-shoota — APPROVED** [review]

**Pull request merged**